### PR TITLE
metrics show: add note about xpath overwriting xpath defined in stage file

### PR DESCRIPTION
**Pull request by pared**
Related to iterative/dvc#2285

### CODE REVIEW

**Comment by pared** [comment]
@shcheklein I think we need to keep this logic. Type and xpath are entangled and we will probably not avoid this logic without some hard assumption, like "`-t` can be used only in `metrics modify`". I will try to make it a little bit more clear.

**Comment by shcheklein** [comment]
@pared I'm not sure I understand this point `like "-t can be used only in metrics modify"`. Right now as far as I understand, documentation is wrong after the last PR was merged. Specifically, `If `-t` is not specified `-x` is ignored.` part. I'm fine to keep the logic if you think it makes sense, but docs should be updated appropriately, and comments in the code.

File: static/docs/commands-reference/metrics_show.md [hunk 1]
```diff
@@ -22,11 +22,18 @@ The optional `path` argument can represent a DVC metric file or a directory. If
 `path` is a directory, recursively search and process all metric files in it
 with the `-R` option.
 
-If metric type is specified (via `-t`) it overrides the type and `--xpath`
-defined (if any) in the corresponding DVC-file (usually, using
-`dvc metrics modify` command). If `-t` is not specified `-x` is ignored. The
-type can be detected by the file extension automatically if the type is
-supported.
+- If metric type (via `-t`) and xpath(`-x`) is specified, it overrides the
+type and xpath
```
Comment (shcheklein) [inline static/docs/commands-reference/metrics_show.md]: extra index that is not needed

File: static/docs/commands-reference/metrics_show.md [hunk 2]
```diff
@@ -22,11 +22,18 @@ The optional `path` argument can represent a DVC metric file or a directory. If
 `path` is a directory, recursively search and process all metric files in it
 with the `-R` option.
 
-If metric type is specified (via `-t`) it overrides the type and `--xpath`
-defined (if any) in the corresponding DVC-file (usually, using
-`dvc metrics modify` command). If `-t` is not specified `-x` is ignored. The
-type can be detected by the file extension automatically if the type is
-supported.
+- If metric type (via `-t`) and xpath(`-x`) is specified, it overrides the
```
Comment (shcheklein) [inline static/docs/commands-reference/metrics_show.md]: as far as I understand it's enough to specify _only_ `-t` to override the xpath (and make effectively empty if it's not provided explicitly)

File: static/docs/commands-reference/metrics_show.md [hunk 3]
```diff
@@ -22,11 +22,18 @@ The optional `path` argument can represent a DVC metric file or a directory. If
...
+type and xpath
+defined(if any) in the corresponding DVC-file (usually, using `dvc metrics
```
Comment (shcheklein) [inline static/docs/commands-reference/metrics_show.md]: extra space is needed before (

File: static/docs/commands-reference/metrics_show.md (hunk 2 above)
Comment (shcheklein) [reply static/docs/commands-reference/metrics_show.md]: oh, I see the comment below

File: static/docs/commands-reference/metrics_show.md (hunk 1 above)
Comment (shcheklein) [inline static/docs/commands-reference/metrics_show.md]: please use `.` at the end

File: static/docs/commands-reference/metrics_show.md [hunk 4]
```diff
@@ -47,6 +54,8 @@ supported.
   value. Should be used if metric file contains multiple numbers and you need to
   get a only one of them. Only single path is allowed. If multiple metric files
   exist in the project, the same parser and path will be applied to all of them.
+  If xpath for particular metric has been set using [`dvc metrics modify`](https://dvc.org/doc/commands-reference/metrics-modify#options)
+  xpath passed in this option will owervrite it, only for current command run.
```
Comment (shcheklein) [inline static/docs/commands-reference/metrics_show.md]: indent

File: static/docs/commands-reference/metrics_show.md (hunk 2 above)
Comment (shcheklein) [reply static/docs/commands-reference/metrics_show.md]: are specified

Comment (shcheklein) [reply static/docs/commands-reference/metrics_show.md]: the override

File: static/docs/commands-reference/metrics_show.md [hunk 5]
```diff
@@ -22,11 +22,18 @@ The optional `path` argument can represent a DVC metric file or a directory. If
...
-If metric type is specified (via `-t`) it overrides the type and `--xpath`
-defined (if any) in the corresponding DVC-file (usually, using
-`dvc metrics modify` command). If `-t` is not specified `-x` is ignored. The
-type can be detected by the file extension automatically if the type is
-supported.
+- If metric type (via `-t`) and xpath(`-x`) is specified, it overrides the
+type and xpath
+defined(if any) in the corresponding DVC-file (usually, using `dvc metrics
+modify` command).
+- If metric type is specified and xpath is not, type will overwrite the type
```
Comment (shcheklein) [inline static/docs/commands-reference/metrics_show.md]: let's use `xpath` and `type` everywhere

Comment (shcheklein) [reply static/docs/commands-reference/metrics_show.md]: let's mention (via `-t`)

File: static/docs/commands-reference/metrics_show.md [hunk 6]
```diff
@@ -22,11 +22,18 @@ The optional `path` argument can represent a DVC metric file or a directory. If
...
+specified in DVC-file, no xpath will be considered, which will result in
+displaying all metrics from file.
+- If metric type is not specified and xpath is, DVC will try to read type from
+associated DVC-file and apply xpath provided in `-x`.
```
Comment (shcheklein) [inline static/docs/commands-reference/metrics_show.md]: `associated` is probably not the right term.

just `from the DVC-file`

**Pull request merged**